lib/amazon_ssa_support/rolling_s3_outputter.rb

```diff
@@ -0,0 +1,110 @@
+require "log4r/outputter/fileoutputter"
+require "log4r/staticlogger"
+
+module Log4r
+  
+  class RollingS3Outputter < FileOutputter
+    attr_reader :count, :maxsize
+
+    def initialize(_name, hash={})
+      @count           = 0
+      @aws_args        = hash[:aws_args]
+      @evm_bucket_name = @aws_args[:evm_bucket]
+      @s3_log_prefix   = File.join(@aws_args[:log_prefix], @aws_args[:extractor_id])
+      @evm_bucket      = AmazonSsaSupport::EvmBucket.get(@aws_args)
+            
+      _maxsize = (hash[:maxsize] or hash['maxsize']).to_i
+      if _maxsize.class != Fixnum
+        raise TypeError, "Argument 'maxsize' must be an Fixnum", caller
+      end
+      if _maxsize == 0
+        raise TypeError, "Argument 'maxsize' must be > 0", caller
+      end
+      @maxsize = _maxsize
+      @datasize = 0
+      
+      super(_name, hash.merge({:create => true, :trunc => true}))
+    end
+    
+    def flush
+      roll
+    end
+
+    private
+
+    def write(data)
+      # we have to keep track of the file size ourselves - File.size doesn't
+      # seem to report the correct size when the size changes rapidly
+      @datasize += data.size + 1 # the 1 is for newline
+      super
+      roll if @datasize > @maxsize
+    end
+    
+    def newS3OblectKey
```

Comment:
change name to new_s3_object_key - or - better yet - s3_object_key - no reason for the "new".  Also there is no such thing as an "Oblect" - that is "Object".  :-)